test/test_eval.py

```diff
@@ -358,6 +358,44 @@ def test_extractive_qa_eval_multiple_queries(reader, retriever_with_docs, tmp_pa
     assert metrics["Retriever"]["ndcg"] == 0.5
 
 
+@pytest.mark.parametrize("retriever_with_docs", ["elasticsearch"], indirect=True)
+@pytest.mark.parametrize("document_store_with_docs", ["elasticsearch"], indirect=True)
+def test_extractive_qa_labels_with_filters(reader, retriever_with_docs, tmp_path):
+    labels = [
+        # MultiLabel with filter that selects only the document about Carla
+        MultiLabel(labels=[Label(query="What's her name?", answer=Answer(answer="Carla", offsets_in_context=[Span(11, 16)]),
+            document=Document(id='a0747b83aea0b60c4b114b15476dd32d', content_type="text", content='My name is Carla and I live in Berlin'),
+            is_correct_answer=True, is_correct_document=True, origin="gold-label", filters={"name":["filename1"]})]),
+        # MultiLabel with filter that selects only the document about Christelle
+        MultiLabel(labels=[Label(query="What's her name?", answer=Answer(answer="Christelle", offsets_in_context=[Span(11, 20)]),
+            document=Document(id='4fa3938bef1d83e4d927669666d0b705', content_type="text", content='My name is Christelle and I live in Paris'),
+            is_correct_answer=True, is_correct_document=True, origin="gold-label", filters={"name":["filename3"]})])
+    ]
+
+    pipeline = ExtractiveQAPipeline(reader=reader, retriever=retriever_with_docs)
+    eval_result = pipeline.eval(
+        labels=labels,
+        params={"Retriever": {"top_k": 5}},
+    )
+
+    metrics = eval_result.calculate_metrics()
+
+    reader_result = eval_result["Reader"]
+    retriever_result = eval_result["Retriever"]
+
+    # The same query but with two different filters and thus two different answers is answered correctly in both cases.
+    assert reader_result[reader_result['rank'] == 1]["answer"].iloc[0] in reader_result[reader_result['rank'] == 1]["gold_answers"].iloc[0]
+    assert retriever_result[retriever_result['rank'] == 1]["document_id"].iloc[0] in retriever_result[retriever_result['rank'] == 1]["gold_document_ids"].iloc[0]
+    assert metrics["Reader"]["exact_match"] == 1.0
+    assert metrics["Reader"]["f1"] == 1.0
+    assert metrics["Retriever"]["mrr"] == 1.0
+    assert metrics["Retriever"]["recall_multi_hit"] == 1.0
+    assert metrics["Retriever"]["recall_single_hit"] == 1.0
+    assert metrics["Retriever"]["precision"] == 1.0
+    assert metrics["Retriever"]["map"] == 1.0
+    assert metrics["Retriever"]["ndcg"] == 1.0
+
+
```

Comment:
I'd suggest to add another test in `test_schema` that validates that the aggregation logic in multilabel (`_aggregate_labels`)  works as intended with filters.